### DISCUSSION TITLE
Nested collections

### DISCUSSION THREAD

**Discussion by horuskol** [tighten/jigsaw#574] [Ideas]
Hi, I'm aware there is already an issue for nested collections at https://github.com/tighten/jigsaw/issues/160, and that it has been tagged as 2.0

I wonder if you would be open to a PR that brought a solution?

Here's an example directory structure and configuration, with the resulting output.

```
source /
  _posts /
    20210101-first-blog-post.blade.md
    202102 /
      01-a-blog-post-in-february.blade.md
      02-another-blog-post-in-february.blade.md
    20210301-blog-post-with-sections /
      index.blade.md
      _sections /
        01-first-section.blade.md
        02-second-section.blade.md
```

```php
return [
  'collections' => [
    'posts' => 'blog/{date|Y-m-d}/{-title}
    'collections' => [
      'sections' => [
        'path' => '{-filename}',
      ],
    ],
  ],
];
```

Resulting output:

```
build /
  blog /
    2021-01-01 /
      first-blog-post-title /
        index.html
    2021-02-01 /
      a-blog-post-in-february-title /
        index.html
    2021-02-10 /
      another-blog-post-in-february-title /
        index.html
    2021-03-01 /
      blog-post-with-sections-title /
        index.html
        01-first-section /
          index.html
        02-second-section /
          index.html
```

Some notes:

1. The default behaviour of including files in subfolders as part of the collection must remain;
2. If a subfolder name (`_sections`) matches the subcollection name (`sections`), these files should not be part of the parent collection;
3. The subcollection should have access to the title and url of its parent.

**Comment by bakerkretzmar** [comment]
Whether we would decide to add this to v1 would depend a lot on the implementation and the maintenance burden it could add, but I for one am very curious to see how you got this working 😅

**Reply by horuskol** [reply]
To be honest, I haven't dug too far into this yet - it's something I've bumped on a couple of times, though. I'll be happy to start going a bit deeper.

Do you have any feedback about the behaviour/output I outlined above?

**Comment by horuskol** [comment]
Sorry there hasn't been any updates - been caught up with other things and haven't been able to find the time to deep dive and make a go of it.
